### DISCUSSION TITLE
flask-socketio does not work

### DISCUSSION THREAD

**Discussion by alitrack** [bigskysoftware/htmx#714] [Q&A]
I tried fastapi and flask-sockets, they works

 tried htmx(1.6.0 and 1.6.1) with flask-socketio, does not work,

when async_mode
- eventlet gives 404
- gevent  `WebSocket is already in CLOSING or CLOSED state.`, but `client.py` works.

the script is here
https://github.com/alitrack/htmx_ws

**Comment by eliaswestonfarber** [comment]
Ok so here is the thing... htmx needs a real websocket and socketio is not a true websocket try flask-sock instead https://github.com/miguelgrinberg/flask-sock/issues
